BUILDING.md

```diff
@@ -126,6 +126,17 @@ To run the tests:
 $ make test
 ```
 
+To run the tests and generate code coverage reports:
```

Comment:
Hey, thanks for writing it up!

From my limited experience with `make coverage`, it can have some quite surprising behavior. I'd suggest noting the following caveats in this section of the docs. Whether to note some or all of them is up to you!

- It doesn't work on macOS (I think? /cc @Trott)
- The `lib/` directory is overwritten during `make coverage`, and the original copied to `lib_/`. I'd always suggest checking in the `lib/` directory to Git before running coverage reports in case something awry happens. See https://github.com/nodejs/node/blob/86e7c61a070d056daec9a756b86aece9ec30c2a8/Makefile#L151-L152.
- The initial `make coverage` run downloads some tools to the project root folder that are not `.gitignore`d, which can be a nuisance.
- All effects of `make coverage` (`lib/` renaming, downloaded tools, raw and processed output) can be reverted with `make coverage-clean`
- `make coverage` both builds and runs (i.e. `make` + `make test`); no way to split them properly unfortunately (one could `coverage-build` first, but when `coverage-test`ing the JS files will get instrumented again, and the `node` binary rebuilt based on the instrumented JS files).

Reply:
Thanks for your review @TimothyGu. I will document `make coverage-clean` too. I'm also wondering if we should handle setting clean as an option on the command line i.e. `make coverage --clean`. I think this would be good but `coverage-clean` also removes the `coverage/` dir with the reports on so if passing the option `--clean` then we would want to leave the `coverage/` dir intact. I don't think I see this as an issue as long as that dir is added to `.gitignore`. What do you think?

Reply:
Also, I just noticed that on the install of istanbul-merge and nyc npm generates a `package-lock.json` which isn't removed by `make coverage-clean`.

Reply:
> I'm also wondering if we should handle setting clean as an option on the command line i.e. `make coverage --clean=true`

That would be nice, but GNU Make doesn't support doing that AFAIK.

> Also, I just noticed that on the install of istanbul-merge and nyc, npm generates a `package-lock.json` which isn't removed by `make coverage-clean`.

Good catch! You can submit another pull request to change that if you'd like to, or if not please file an issue about it so we don't forget :)

Reply:
Great I will raise two other PRs and yes you're right my syntax was wrong on `make coverage CLEAN=true` :)

Reply:
I'll wait to this to land before submitting the PR for handling passing clean as an option so that I can update the docs for that.